<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-4-g7f85cfe  Build Version: 1.6.7995.38578
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot
  name="roboturdf">
  <link
    name="base_link">
    <inertial>
      <origin
        xyz="-0.00991569326653178 -0.00913564327655161 0.0120638268093608"
        rpy="0 0 0" />
      <mass
        value="0.0424207954539648" />
      <inertia
        ixx="2.16321488293071E-05"
        ixy="9.94980765555521E-16"
        ixz="8.95804033158309E-17"
        iyy="2.15757798258962E-05"
        iyz="2.411565593253E-16"
        izz="3.74667249407743E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf_tutorial/meshes/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf_tutorial/meshes/base_link.STL" />
      </geometry>
    </collision>
  </link>
  <link
    name="link1">
    <inertial>
      <origin
        xyz="3.5593E-05 0.014619 0"
        rpy="0 0 0" />
      <mass
        value="0.0082741" />
      <inertia
        ixx="6.5846E-07"
        ixy="-4.5775E-09"
        ixz="-1.5516E-23"
        iyy="5.1597E-07"
        iyz="-6.0176E-23"
        izz="9.7842E-07" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf_tutorial/meshes/link1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf_tutorial/meshes/link1.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint1"
    type="fixed">
    <origin
      xyz="-0.010043 -0.0092626 0.0055839"
      rpy="1.5708 0 0" />
    <parent
      link="base_link" />
    <child
      link="link1" />
    <axis
      xyz="0 0 0" />
  </joint>
  <link
    name="link2">
    <inertial>
      <origin
        xyz="5.2042E-18 0.0029752 -8.6736E-19"
        rpy="0 0 0" />
      <mass
        value="0.00041234" />
      <inertia
        ixx="1.5141E-09"
        ixy="1.2535E-24"
        ixz="-2.068E-23"
        iyy="3.3477E-08"
        iyz="3.4431E-25"
        izz="3.2482E-08" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf_tutorial/meshes/link2.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf_tutorial/meshes/link2.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint2"
    type="continuous">
    <origin
      xyz="0.005207 0.02975 0"
      rpy="-3.1416 -1.4454 3.1416" />
    <parent
      link="link1" />
    <child
      link="link2" />
    <axis
      xyz="0 1 0" />
  </joint>
  <link
    name="link3">
    <inertial>
      <origin
        xyz="-0.0014847 0.010393 -9.4182E-05"
        rpy="0 0 0" />
      <mass
        value="0.0079558" />
      <inertia
        ixx="2.5184E-06"
        ixy="8.9278E-08"
        ixz="-2.2474E-08"
        iyy="3.8924E-06"
        iyz="1.9527E-08"
        izz="3.589E-06" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf_tutorial/meshes/link3.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf_tutorial/meshes/link3.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint3"
    type="fixed">
    <origin
      xyz="0 0.004 -6.3975E-05"
      rpy="0 -0.005563 0" />
    <parent
      link="link2" />
    <child
      link="link3" />
    <axis
      xyz="0 0 0" />
  </joint>
  <link
    name="link4">
    <inertial>
      <origin
        xyz="3.5593E-05 0.014619 1.3878E-17"
        rpy="0 0 0" />
      <mass
        value="0.0082741" />
      <inertia
        ixx="6.5846E-07"
        ixy="-4.5775E-09"
        ixz="2.5725E-22"
        iyy="5.1597E-07"
        iyz="8.2718E-23"
        izz="9.7842E-07" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf_tutorial/meshes/link4.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf_tutorial/meshes/link4.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint4"
    type="fixed">
    <origin
      xyz="-0.038256 0.021532 0.0016152"
      rpy="-3.1416 -0.8716 1.5708" />
    <parent
      link="link3" />
    <child
      link="link4" />
    <axis
      xyz="0 0 0" />
  </joint>
  <link
    name="link5">
    <inertial>
      <origin
        xyz="6.9389E-18 0.00289 6.9389E-18"
        rpy="0 0 0" />
      <mass
        value="0.00041234" />
      <inertia
        ixx="1.5141E-09"
        ixy="1.7513E-23"
        ixz="-6.8953E-24"
        iyy="3.3477E-08"
        iyz="-2.068E-25"
        izz="3.2482E-08" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf_tutorial/meshes/link5.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf_tutorial/meshes/link5.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint5"
    type="continuous">
    <origin
      xyz="0.005207 0.029835 0"
      rpy="0 -0.26894 0" />
    <parent
      link="link4" />
    <child
      link="link5" />
    <axis
      xyz="0 1 0" />
  </joint>
  <link
    name="link6">
    <inertial>
      <origin
        xyz="-8.2845E-05 0.001 -0.024692"
        rpy="0 0 0" />
      <mass
        value="0.0036791" />
      <inertia
        ixx="1.6299E-06"
        ixy="-1.5882E-22"
        ixz="-3.0229E-09"
        iyy="1.9249E-06"
        iyz="1.0919E-22"
        izz="2.9738E-07" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf_tutorial/meshes/link6.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf_tutorial/meshes/link6.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint6"
    type="fixed">
    <origin
      xyz="-5.9375E-05 0.0059147 0"
      rpy="3.1416 5.0333E-05 0" />
    <parent
      link="link5" />
    <child
      link="link6" />
    <axis
      xyz="0 0 0" />
  </joint>

  <ros2_control name="RRBotSystemPositionOnly" type="system">
    <hardware>
      <plugin>ros2_control_demo_example_1/RRBotSystemPositionOnlyHardware</plugin>
        <param name="example_param_hw_start_duration_sec">0</param>
        <param name="example_param_hw_stop_duration_sec">3.0</param>
        <param name="example_param_hw_slowdown">100</param>
    </hardware>
    <joint name="joint2">
      <command_interface name="position">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <state_interface name="position"/>
    </joint>
    <joint name="joint5">
      <command_interface name="position">
        <param name="min">-1</param>
        <param name="max">1</param>
      </command_interface>
      <state_interface name="position"/>
    </joint>

  </ros2_control>

</robot>